app/controllers/dashboards_controller.rb

```diff
@@ -22,9 +31,29 @@ def export
     redirect_to dashboards_path, alert: e.message
   end
 
+  def push
+    version = Version.preview_version
+
+    if version.blank?
+      flash.alert = 'No preview version available'
+    else
+      pv = Version.create(number: version.number, production: true, user: current_user)
+
+      if pv.persisted?
+        flash.notice = 'Production data updated'
+      else
+
+        flash.alert = "Production data not updated: #{version.inspect} #{pv.errors.inspect}"
+      end
+    end
+
+    redirect_to dashboards_path
+  end
+
   private
 
   def csv_model(csv_type)
+    return Institution if csv_type == 'Institution'
```

Comment:
yea, and don't shy away from creating your own error class that inherits from StandardError when doing rescues. It's a lot better than rescuing from StandardError when that need arises.